gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/KspExtension.kt

```diff
@@ -18,19 +18,24 @@
 package com.google.devtools.ksp.gradle
 
 import org.gradle.api.GradleException
+import org.gradle.api.Project
 import org.gradle.api.file.ConfigurableFileCollection
+import org.gradle.api.provider.Provider
 import org.gradle.process.CommandLineArgumentProvider
+import javax.inject.Inject
 
-abstract class KspExtension {
-    internal val apOptions = mutableMapOf<String, String>()
-    internal val commandLineArgumentProviders = mutableListOf<CommandLineArgumentProvider>()
-    internal val excludedProcessors = mutableSetOf<String>()
+abstract class KspExtension @Inject constructor(project: Project) {
+    internal val apOptions = project.objects.mapProperty(String::class.java, String::class.java)
```

Comment:
Ideally this should have `finalizeValueOnRead` as well, but since it's exposed as `arguments`, it's not safe if consumers are reading the values.